manifests/designate/services.pp

```diff
@@ -0,0 +1,91 @@
+# Configures the basic requirements for designate and the designate-services
+class ntnuopenstack::designate::services {
+  require ::ntnuopenstack::designate::dbconnection
+  include ::ntnuopenstack::designate::firewall::api
+
+  # Common / base configuration
+  $rabbitservers = lookup('profile::rabbitmq::servers', {
+    'value_type'    => Variant[Array[String], Boolean],
+    'default_value' => false,
+  })
+
+  if ($rabbitservers) {
+    $ha_transport_conf = {
+      rabbit_ha_queues    => true,
+      amqp_durable_queues => true,
+    }
+  } else {
+    $ha_transport_conf = {}
+  }
+
+  $transport_url = lookup('ntnuopenstack::transport::url')
+
+  class { '::designate':
+    default_transport_url => $transport_url,
+    *                     => $ha_transport_conf,
+  }
+
+  # Coordination
+  $zookeeper_servers = lookup('profile::zookeeper::servers', {
+    'value_type'    => Hash[String, Stdlib::IP::Address::Nosubnet],
+  })
+  $zookeeper_urls = values($zookeeper_servers).map | $server | {
+    "${server}:2181"
+  }
+  class { '::designate::coordination':
+    backend_url => "'zookeeper://${join($zookeeper_urls, ',')}'",
+  }
+  # Until we update to >= https://review.opendev.org/c/openstack/puppet-oslo/+/917759
+  ensure_packages('python3-kazoo', {
+    'ensure' => 'present'
+  })
+
+  # designate-api
+  $api_port = lookup('ntnuopenstack::designate::api::port')
+
+  class { '::designate::api':
+    auth_strategy    => 'keystone',
+    enable_api_v2    => true,
+    enable_api_admin => true,
+    service_name     => 'httpd',
+  }
+
+  class { '::designate::wsgi::apache':
+    access_log_format => 'forwarded',
+    workers           => 2,
```

Comment:
Dette burde erfaringsmessig kunne overstyres. To WSGI-workers kan fort bli problematisk ved "høy" last og mange forespørsler. Foreslår å la den være default (den defaulter til $facts['os_workers']) også kan vi eventuelt overkjøre den med hiera etter behov. Det er det vi gjør med alt annet

Reply:
Løst i https://github.com/ntnusky/ntnuopenstack/pull/101/commits/bbdb09dedc0b83fff00ca99d0381dc69a5284d11